src_new/controllers/uas.js

```diff
@@ -0,0 +1,441 @@
+            }
+
+            setTimeout(function() {
+              UTIL.log("PostTimeout.. back in display function");
+              UTIL.forEachOnObject(refThis.slotsMap, function(key, slot) {
+                  refThis.findWinningBidIfRequired_Display(key, slot);
+              });
+              refThis.processDisplayCalledSlot(theObject, originalFunction, arg);
+            }, CONFIG.getTimeout());
+
+            break;
+            // call the original function now
+        case CONSTANTS.SLOT_STATUS.TARGETING_ADDED:
+            refThis.updateStatusAndCallOriginalFunction_Display(
+                "As DM processing is already done, Calling original display function with arguments",
+                theObject,
+                originalFunction,
+                arg
+            );
+            break;
+
+        case CONSTANTS.SLOT_STATUS.DISPLAYED:
+            refThis.updateStatusAndCallOriginalFunction_Display(
+                "As slot is already displayed, Calling original display function with arguments",
+                theObject,
+                originalFunction,
+                arg
+            );
+            break;
+    }
+}
+
+/* start-test-block */
+exports.displayFunctionStatusHandler = displayFunctionStatusHandler;
+/* end-test-block */
+
+function initiateDisplay(win) {
+  win.Phoenix.registerPreDisplayHandler(function(originalDisplay, divId){
+    var phoenixObj = window.Phoenix || {};
+
+    /* istanbul ignore next */
+    refThis.updateSlotsMapFromPhoenixSlots(phoenixObj.getSlots(), [divId], true);
+
+    /* istanbul ignore next */
+    refThis.displayFunctionStatusHandler(getStatusOfSlotForDivId(divId), {}, originalDisplay, [divId]);
+
+    var statusObj = {};
+    statusObj[CONSTANTS.SLOT_STATUS.CREATED] = "";
+
+    refThis.forQualifyingSlotNamesCallAdapters(getSlotNamesByStatus(statusObj), [divId], false);
+
+    /* istanbul ignore next */
+    setTimeout(function() {
+      UTIL.realignVLogInfoPanel(divId);
+      bidManager.executeAnalyticsPixel();
+    }, 2000 + CONFIG.getTimeout());
+
+  });
+}
+/* start-test-block */
+exports.initiateDisplay = initiateDisplay;
+/* end-test-block */
+
+function callJsLoadedIfRequired(win) { // TDD, i/o : done
+    if (UTIL.isObject(win) && UTIL.isObject(win.PWT) && UTIL.isFunction(win.PWT.jsLoaded)) {
+        win.PWT.jsLoaded();
+        return true;
+    }
+    return false;
+}
+/* start-test-block */
+exports.callJsLoadedIfRequired = callJsLoadedIfRequired;
+/* end-test-block */
+
+function createPhoenixNamespace(win){
+	// if Phoenix.isJSLoaded is undefined then it means,
+	// this is first time our JS is loaded on page
+	var Phoenix = win.Phoenix;
+	if( UTIL.isUndefined(Phoenix) || UTIL.isUndefined(Phoenix.isJSLoaded) ){
+		win.Phoenix = new PHOENIX(CONFIG.getTimeout());
+    refThis.initiateDisplay(win);
+	}
+	return win.Phoenix;
+}
+/* start-test-block */
+exports.createPhoenixNamespace = createPhoenixNamespace;
+/* end-test-block */
+
+function initPhoenix(win){
```

Comment:
this code should go into phoenix.js

Reply:
else it will be duplication, OpenWrap can use phoenix EQ to add preDisplayHandler